### ISSUE TITLE
Left Recursion within Property Grammar

### ISSUE THREAD

**Issue by vihanb** [cheddar-lang/Cheddar#5]
Not exactly the main cause of this but left recursion appears to occur when literals/expressions are parsed with properties. Possibly caused by a non-restrictive parenthesized expression rule. Examples are:

```js
"foo".bar
(a+b).c
```

As you can see each of these have expressions as their values. This could be possibly combated using:

```
@head = (literal | "(" expression ")" | identifier) @body
```

---

This was removed in `develop/v0.0.1xx` to avoid the bug rippling into larger problems. While this may be taken out, it is not to be ignored and is scheduled for a fix by v0.2-alpha

**Comment by vihanb** [comment]
I change my mind. This is very important so I'm targetting targeting it for v1.0.0

**Comment by vihanb** [comment]
Fixed in d8f58a2e2ae337698bd5d0b34abb3f26b2c3665e

**Comment by schas002** [comment]
Finally! :facepalm:
